hat/assets/js/apps/Iaso/domains/instances/compare/components/CompareInstanceLogs.tsx

```diff
@@ -24,92 +34,107 @@ type RouterCustom = {
-
+    const [instanceLogInfos, setInstanceLogInfos] = useState({
+        logA: {
+            ...logInfos,
+        },
+        logB: {
+            ...logInfos,
+        },
+    });
     const [logAInitialValue, setLogAInitialValue] = useState<
         number | undefined
     >(undefined);
     const [logBInitialValue, setLogBInitialValue] = useState<
         number | undefined
     >(undefined);
 
-    const { instanceIds: instanceId } = params;
-
-    const { data: instanceLogsDropdown, isFetching: isFetchingInstanceLogs } =
-        useGetInstanceLogs(instanceId);
-
     const handleChange = (key, value) => {
         const newParams = {
             ...params,
             [key]: value,
         };
-        dispatch(redirectTo(baseUrls.compareInstanceLogs, newParams));
+        dispatch(redirectToReplace(baseUrls.compareInstanceLogs, newParams));
     };
 
     useEffect(() => {
         if (instanceLogsDropdown) {
-            if (params.logA === undefined && params.logB === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logA: instanceLogsDropdown[0]?.value,
-                    logB: instanceLogsDropdown[1]?.value,
-                };
-                dispatch(
-                    redirectTo(baseUrls.compareInstanceLogs, defaultParams),
-                );
-            } else if (params.logA === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logA: instanceLogsDropdown[0]?.value,
-                };
-                dispatch(
-                    redirectTo(baseUrls.compareInstanceLogs, defaultParams),
-                );
-            } else if (params.logB === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logB: instanceLogsDropdown[1]?.value,
-                };
-
+            const defaultParams: DefaultParams = {
+                ...params,
+            };
+            if (!params.logA && instanceLogsDropdown[0]?.value) {
+                defaultParams.logA = instanceLogsDropdown[0]?.value.toString();
+            }
+            if (!params.logB && instanceLogsDropdown[1]?.value) {
+                defaultParams.logB = instanceLogsDropdown[1]?.value.toString();
+            }
+            if (
+                (!params.logA && instanceLogsDropdown[0]?.value) ||
+                (!params.logB && instanceLogsDropdown[1]?.value)
+            ) {
                 dispatch(
-                    redirectTo(baseUrls.compareInstanceLogs, defaultParams),
+                    redirectToReplace(
+                        baseUrls.compareInstanceLogs,
+                        defaultParams,
+                    ),
                 );
             }
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [instanceLogsDropdown, params]);
-
+    }, [instanceLogsDropdown]);
```

Comment:
Why not including `params`? As a general rule we should always include all deps as ignoring some can cause unexpected bugs.
If including params was causing an infinite loop, maybe we can look for another solution.